packages/metablocks/src/api.js

```diff
@@ -25,8 +39,257 @@ const createUniverse = async ({
+      program,
+      usersKey,
+      mintKey,
+      universeKey,
+    });
+    const transaction = new Transaction();
+    transaction.add(depositNftInstruction);
+
+    return await program.provider.send(transaction, []);
+  } catch (e) {
+    console.log("from api", e);
+    throw e;
+  } finally {
+  }
+};
+
+const transferReceiptNftToUserV1 = async ({
+  connection,
+  wallet,
+  usersKey,
+  mintKey,
+  universeKey,
+  url,
+  isReceiptMasterEdition,
+}) => {
+  try {
+    const program = getMetaBlocksProgram(connection, wallet);
+    const transferReceiptNftToUserInstruction =
+      await getTransferReceiptNftToUserInstruction({
+        program,
+        usersKey,
+        mintKey,
+        universeKey,
+        url,
+        isReceiptMasterEdition,
+      });
+    const transaction = new Transaction();
+    transaction.add(transferReceiptNftToUserInstruction);
+
+    const tx = await program.provider.send(transaction, []);
+    return tx;
+  } catch (e) {
+    console.log("from api", e);
+    throw e;
+  } finally {
+  }
+};
+
+const groupedDepositNftV1 = async ({
+  connection,
+  wallet,
+  usersKey,
+  mintKey,
+  universeKey,
+  receiptNftUrl,
+  isReceiptMasterEdition,
+}) => {
+  try {
+    const program = getMetaBlocksProgram(connection, wallet);
+
+    const {
+      initReceiptMint: { initReceiptMintArgs, initReceiptMintAccounts },
+      initDepositNft: { initDepositNftArgs, initDepositNftAccounts },
+      depositNft: { depositNftArgs, depositNftAccounts },
+      transferReceiptNft: {
+        transferReceiptNftArgs,
+        transferReceiptNftAccounts,
+      },
+    } = await computeGroupedDepositNftParams({
+      usersKey,
+      mintKey,
+      universeKey,
+      receiptNftUrl,
+      isReceiptMasterEdition,
+    });
+
+    const initReceiptMintInstruction = program.instruction.initReceiptMintV1(
+      initReceiptMintArgs,
+      {
+        accounts: initReceiptMintAccounts,
+      }
+    );
+
+    const initDepositNftInstruction = program.instruction.initDepositNftV1(
+      initDepositNftArgs,
+      {
+        accounts: initDepositNftAccounts,
+      }
+    );
+
+    const transaction1 = new Transaction();
+    transaction1.add(initReceiptMintInstruction);
+    transaction1.add(initDepositNftInstruction);
+    const tx1 = await program.provider.send(transaction1, []);
+
+    const depositNftInstruction = program.instruction.depositNftV1(
+      depositNftArgs,
+      {
+        accounts: depositNftAccounts,
+      }
+    );
+    const transferReceiptNftToUserInstruction =
+      program.instruction.transferReceiptNftToUserV1(transferReceiptNftArgs, {
+        accounts: transferReceiptNftAccounts,
+      });
+
+    const transaction2 = new Transaction();
+    transaction2.add(depositNftInstruction);
+    transaction2.add(transferReceiptNftToUserInstruction);
+    const tx2 = await program.provider.send(transaction2, []);
+
+    return { tx1, tx2 };
```

Comment:
These are two txns -> will the user will need to sign twice?

Reply:
This will be handled in this story - https://app.shortcut.com/meta-blocks/story/163/kyraa-sign-all-transaction